app-libs/stf/src/best_energy_helpers.rs

```diff
@@ -0,0 +1,82 @@
+use crate::MerkleProofWithCodec;
+use binary_merkle_tree::merkle_proof;
+use codec::Encode;
+use simplyr_lib::Order;
+use sp_core::H256;
+use sp_runtime::traits::{Hash, Keccak256};
+
+/// Gets the merkle proof of an `actor_id` if it is in the order set.
+pub fn get_merkle_proof_for_actor(
+	actor_id: &str,
+	orders: &[Order],
+) -> Option<MerkleProofWithCodec<H256, Vec<u8>>> {
+	let leaf_index = get_leaf_index_for_actor(actor_id, orders)?;
+	Some(merkle_proof::<Keccak256, _, _>(orders.iter().map(Encode::encode), leaf_index).into())
```

Comment:
`.map(Encode::encode)` is shorthand for `map(|o| o.encode()`.

Note: we omit the `collect::<Vec<_>>` here because we can directly add an iterator into the `merkle_proof` function - this saves use `O(n)` computations.